kolibri/core/auth/test/test_api.py

```diff
@@ -987,6 +987,21 @@ def post_resetsettings():
             response = post_resetsettings()
             self.assertDictContainsSubset(mappings[setting], response.data)
 
+    def test_validation_for_incompatible_settings(self):
```

Comment:
There is another case here, where the two values are set in sequential requests. I would suggest making a second test case which does two requests, the first one setting `"learner_can_login_with_no_password": "true",` and the second setting `"learner_can_edit_password": "true",` if the validation is working correctly, then this should produce a 400.

I think it will probably produce a 500 because of an unhandled IntegrityError however.

Reply:
I missed this. My bad. I have added the test case. Also added another case where if only `"learner_can_login_with_no_password": "true"` is set, a 400 is checked for since `"learner_can_edit_password": "true"` is a default value.